server/api/launcher.py

```diff
@@ -175,6 +175,60 @@ def launch(
 
         return self._wrap_run_result(runtime, result, run, err=last_err)
 
+    def _enrich_run(
+        self,
+        runtime,
+        run,
+        handler=None,
+        project_name=None,
+        name=None,
+        params=None,
+        inputs=None,
+        returns=None,
+        hyperparams=None,
+        hyper_param_options=None,
+        verbose=None,
+        scrape_metrics=None,
+        out_path=None,
+        artifact_path=None,
+        workdir=None,
+        notifications: list[mlrun.model.Notification] = None,
+        state_thresholds: Optional[dict[str, int]] = None,
+    ):
+        run = super()._enrich_run(
+            runtime=runtime,
+            run=run,
+            handler=handler,
+            project_name=project_name,
+            name=name,
+            params=params,
+            inputs=inputs,
+            returns=returns,
+            hyperparams=hyperparams,
+            hyper_param_options=hyper_param_options,
+            verbose=verbose,
+            scrape_metrics=scrape_metrics,
+            out_path=out_path,
+            artifact_path=artifact_path,
+            workdir=workdir,
+            notifications=notifications,
+            state_thresholds=state_thresholds,
+        )
+        if runtime._get_db():
+            project = runtime._get_db().get_project(run.metadata.project)
+            if project.spec.default_function_node_selector:
+                mlrun.utils.logger.debug(
+                    "Enriching run node selector from project",
+                    project_name=run.metadata.project,
+                    project_node_selector=project.spec.default_function_node_selector,
+                )
+                run.spec.node_selector = mlrun.utils.helpers.merge_with_precedence(
+                    project.spec.default_function_node_selector,
+                    runtime.spec.node_selector,
+                )
```

Comment:
can add a "pre_run_enrichment" and explain in a sentence that we have decided to do the enrichment on run to avoid dirty-ing the runtime itself as the label selector is per-system